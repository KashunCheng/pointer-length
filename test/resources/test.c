//{"func": "test"}
int test(int b, int* a)
{
  int r = 0;
  for(int i = 0; i < b; i++)
  {
    r += a[i];
  }
  return r;
}

int main(int argc, char** argv) {
  int a[5] = {1, 2, 3, 4, 5};
  int b = 5;
  int result = test(b, a);
  return result;
}