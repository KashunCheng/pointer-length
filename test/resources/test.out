ValueNumbering: test
   store i32 %a, i32* %a.addr, align 4    1 = 1
   store i32 %b, i32* %b.addr, align 4    2 = 2
   %0 = load i32, i32* %a.addr, align 4   1 = 1
   %1 = load i32, i32* %b.addr, align 4   2 = 2
   %add = add nsw i32 %0, %1              3 = 1 add 2
   store i32 %add, i32* %c, align 4       3 = 3
   %2 = load i32, i32* %b.addr, align 4   2 = 2
   %3 = load i32, i32* %c, align 4        3 = 3
   %add1 = add nsw i32 %2, %3             4 = 2 add 3
   store i32 %add1, i32* %d, align 4      4 = 4
   %4 = load i32, i32* %a.addr, align 4   1 = 1
   %5 = load i32, i32* %b.addr, align 4   2 = 2
   %add2 = add nsw i32 %4, %5             3 = 1 add 2 (redundant)
   store i32 %add2, i32* %e, align 4      3 = 3
   %6 = load i32, i32* %b.addr, align 4   2 = 2
   %7 = load i32, i32* %e, align 4        3 = 3
   %add3 = add nsw i32 %6, %7             4 = 2 add 3 (redundant)
   store i32 %add3, i32* %f, align 4      4 = 4
   %8 = load i32, i32* %f, align 4        4 = 4
